flamingpy/codes/graphs/egraph.py

```diff
@@ -98,6 +98,198 @@ def macronize(can_graph, pad_boundary=False, disp=0.1):
     return macro_graph
 
 
+def RREform_mod2(M, max_cols=None):
+    """Puts a binary matrix into Row Reduced Echelon form modulo 2 up to a
+    maximum number of columns given by max_cols.
```

Comment:
I suggest a name and description in active voice.
```suggestion
def reduce_RREform_mod2(M, max_cols=None):
    """Put a binary matrix into row reduced echelon form modulo 2, up to a
    maximum number of columns given by max_cols.
```